src/multi_physics/QED/src/utils/progress_bar.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef PICSAR_MULTIPHYSICS_PROGRESS_BAR
+#define PICSAR_MULTIPHYSICS_PROGRESS_BAR
+
+//Should be included by all the src files of the library
+#include "../qed_commons.h"
+
+#include <string>
+#include <iostream>
+
+namespace picsar{
+namespace multi_physics{
+namespace utils{
+
+    /**
+    * A simple progress bar
+    *
+    * @param[in] i current progress index
+    * @param[in] how_many maximum value of the progress index
+    * @param[in] text an optional text to append to the progress bar
+    * @param[in] up_freq frequency at which the progress bar is updated
+    * @param[in] last if true the last character is a new line instead of a carriage return
+    * @param[in] out the std::ostream where the progress bar is drawn
+    */
+    void draw_progress(
+        const int i, const int how_many,
+        const std::string text = "",
+        const int up_freq = 1,
+        bool last=false,
+        std::ostream& out = std::cout)
+        {
+            if (i % up_freq != 0 && i != how_many)
+            return;
+
+            const auto bar_width = 50;
```

Comment:
Would it be better to rename this variable `bar_length`?

Reply:
I agree with you: I've just renamed it...